### PR TITLE
Added proper error for running `next start` on dev build

### LINKED ISSUE #1937: Proper error for running `next start` on dev build
At the moment, we only get this:

screen shot 2017-05-11 at 03 45 48

We need to show a proper error telling the user why it failed.

### PR DESCRIPTION
**Pull request by xuezhma**
Fixes #1937 
Tells user to run `next build` when they run `next start` on dev build.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -33,7 +33,12 @@ export default class Server {
     this.http = null
     this.config = getConfig(this.dir, conf)
     this.dist = this.config.distDir
-    this.buildStats = !dev ? require(join(this.dir, this.dist, 'build-stats.json')) : null
+    try {
+      this.buildStats = !dev ? require(join(this.dir, this.dist, 'build-stats.json')) : null
+    } catch (e) {
+      console.error('No build-stats.json found. Did you forget to run `next build`?')
```
Comment (arunoda) [inline server/index.js]: I'd like to do this in our `bin/*` files.
For an example, do this check here: https://github.com/zeit/next.js/blob/v3-beta/bin/next-start#L55

**Review by arunoda — CHANGES_REQUESTED** [review]

File: server/index.js (hunk 1 above)
Comment (xuezhma) [inline server/index.js]: Looks like `next start` is already checking if a build exists, but the code does the checking is after trying to create a server. I put them in the correct order in my last commit.

**Comment by xuezhma** [comment]
It just occurred to me that checking if a build exists in `/bin/next-start` doesn't cover people with custom servers. Maybe it's better to put the checking in `server/index.js`?

File: server/index.js (hunk 1 above)
Comment (timneutkens) [inline server/index.js]: 👌

**Review by timneutkens — APPROVED** [review]

**Comment by arunoda** [comment]
@xuezhma that's a good point.
Yep. Then that's a good idea. Move the BUILD_ID check from the `bin` to `server/index.js`

**Comment by xuezhma** [comment]
@arunoda Done

**Comment by timneutkens** [comment]
Lets merge this against v3

**Review by timneutkens — APPROVED** [review]

**Comment by timneutkens** [comment]
🎉

**Pull request merged**